src/main/java/at/hannibal2/skyhanni/data/PetAPI.kt

```diff
@@ -35,31 +194,412 @@ object PetAPI {
         "➡",
     )
 
+    @Deprecated(message = "use PetAPI.inPetMenu")
     fun isPetMenu(inventoryTitle: String): Boolean = petMenuPattern.matches(inventoryTitle)
 
     // Contains color code + name and for older SkyHanni users maybe also the pet level
+    @Deprecated(message = "use PetAPI.pet.name")
     var currentPet: String?
         get() = ProfileStorageData.profileSpecific?.currentPet?.takeIf { it.isNotEmpty() }
         set(value) {
             ProfileStorageData.profileSpecific?.currentPet = value
         }
 
-    fun isCurrentPet(petName: String): Boolean = currentPet?.contains(petName) ?: false
+    fun isCurrentPet(petName: String): Boolean = pet?.cleanName?.contains(petName) ?: false
 
     fun getCleanName(nameWithLevel: String): String? {
-        petItemName.matchMatcher(nameWithLevel) {
+        petItemNamePattern.matchMatcher(nameWithLevel) {
             return group("name")
         }
-        neuRepoPetItemName.matchMatcher(nameWithLevel) {
+        neuRepoPetItemNamePattern.matchMatcher(nameWithLevel) {
             return group("name")
         }
 
         return null
     }
 
-    fun getPetLevel(nameWithLevel: String): Int? = petItemName.matchMatcher(nameWithLevel) {
+    @Deprecated(message = "use PetAPI.pet.level")
+    fun getPetLevel(nameWithLevel: String): Int? = petItemNamePattern.matchMatcher(nameWithLevel) {
         group("level").toInt()
     }
 
-    fun hasPetName(name: String): Boolean = petItemName.matches(name) && !ignoredPetStrings.any { name.contains(it) }
+    @Deprecated(message = "use PetAPI.pet.name")
+    fun hasPetName(name: String): Boolean = petItemNamePattern.matches(name) && !ignoredPetStrings.any { name.contains(it) }
+
+// ---
+    @SubscribeEvent
+    fun onWidgetUpdate(event: WidgetUpdateEvent) {
+        if (!event.isWidget(TabWidget.PET)) return
+
+        val newPetLine = petWidgetPattern.firstMatches(event.lines)?.trim() ?: return
+        if (newPetLine == pet?.rawPetName) return
+
+        var newPetData = PetData()
+        var overflowXP = 0.0
+        event.lines.forEach { line ->
```

Comment:
i think it would be better if you make this a normal for loop and use continue instead of return@forEach